src/entryproc.h

```diff
@@ -74,9 +74,9 @@ class EntryProcessor {
     uid_t m_file_uid = (uid_t)-1; ///< The resolved user-id corresponding to the active ProfileEntry.
     gid_t m_file_gid = (gid_t)-1; ///< The resolved group-id corresponding to the active ProfileEntry.
     std::string m_path; ///< The path of the current file to check below a potential m_args.root_path.
-    std::string m_fd_path; ///< A path for safely opening the target file (typically in /proc/self/fd/...).
+    std::string m_safe_path; ///< A path for safely opening the target file (typically in /proc/self/fd/...).
```

Comment:
Why "typically"? Are there circumstances where this is not true?

Reply:
Yes in diff / read-only mode, if /proc is not available, as is documented elsewhere in the code.

Reply:
In that case, is `m_safe_path` not set or is it set to something else?

Reply:
There are three levels and one sub varaint of paths here:

- `m_entry.file` is the full path, which could be including an "alternative root".
- `m_path` is the display path to show for diagnostics, a human readable version, relative to a possible "alternative root".
- `m_safe_path` is an implementation detail that usually refers to a safe variant of the path pointing to /proc.
-  sub variant of `m_safe_path`: an "unsafe" path if `chkstat` considers it okay.

It comes with the heritage of this program that generic code portions are reused for vastly different contexts. I guess it is hard to change this at this stage. We have the feature to operate only in a checking read-only mode where using safe paths is not strictly required. And we have this friendly fallback to allow this read-only mode even if there is no `/proc`.

Basically the codebase guarantees via the logic in `EntryProcess::process()` that `m_safe_path` always is safe given the context that was detected. So it could be argued that it _is_ always safe.

To make this clearer see different possibilities:

- drop the fallback to operate on "unsafe" paths if there is no `/proc` and we only want to "check". I don't believe this feature will ever come into play. `/proc` not being available usually only happens in build environments, as we have seen lately, and I guess no one will enter the build environment to "check" which permissions are missing while `/proc` is not mounted.
- choose a clearer name. Maybe something like `m_api_path` to emphasize the property that it is a variant of the path specific to the APIs used.
- keep it as it is, refering to the reasoning above.